pkg/client/apiutil/dynamicrestmapper_test.go

```diff
@@ -57,53 +58,49 @@ var _ = Describe("Dynamic REST Mapper", func() {
 		})
 
 		It("should reload if not present in the cache", func() {
-			By("reading successfully once")
+			By("reading target successfully once")
 			Expect(callWithTarget()).To(Succeed())
-			Expect(callWithOther()).NotTo(Succeed())
 
-			By("asking for a something that didn't exist previously after adding it to the mapper")
+			By("reading other not successfully")
+			count := 0
 			addToMapper = func(baseMapper *meta.DefaultRESTMapper) {
+				count++
 				baseMapper.Add(targetGVK, meta.RESTScopeNamespace)
-				baseMapper.Add(secondGVK, meta.RESTScopeNamespace)
 			}
-			Expect(callWithOther()).To(Succeed())
-			Expect(callWithTarget()).To(Succeed())
-		})
+			Expect(callWithOther()).To(beNoMatchError())
```

Comment:
Why does this fail on the first attempt now and only works on the second?

Reply:
It was also failing on the first attempt before (see [L62](https://github.com/kubernetes-sigs/controller-runtime/pull/1151/files/93a54481b9d2acb5231e6fd33663b1f9acf1ad4b#diff-515d07e6da6fc55b48a1766c7948ba61L62)) as the `secondGVK` is not present by default (see [L33](https://github.com/kubernetes-sigs/controller-runtime/pull/1151/files/93a54481b9d2acb5231e6fd33663b1f9acf1ad4b#diff-515d07e6da6fc55b48a1766c7948ba61L33))
I just moved down the first `callWithOther` and added `count` to ensure, that the RESTMapper really tries to rediscover for the call with the unknown resource.